### PR TITLE
Fix #386 - Debug export does not include stdClass properties

### LINKED ISSUE #386: Debug export does not include stdClass properties
Example to reproduce:

``` php
$obj = new \stdClass;
$obj->foo = "bar";
$obj->bar = 1234;
$var = \Doctrine\Common\Util\Debug::export($obj, 2);

class a { public $foo; public $bar;}
$obj = new a();
$obj->foo = 'bar';
$obj->bar = 1234;
$var = \Doctrine\Common\Util\Debug::export($obj, 2);
```

### PR DESCRIPTION
**Pull request by ostrolucky**
Fixed thanks to pretty interesting comment posted [12 years ago](http://php.net/manual/en/function.get-object-vars.php#47075)

ref: #386

### CODE REVIEW

File: lib/Doctrine/Common/Util/Debug.php [hunk 1]
```diff
@@ -144,41 +144,26 @@ public static function export($var, $maxDepth)
...
-                    . ($reflProperty->isPrivate() ? ':private' : '')
-                    . ($reflProperty->isProtected() ? ':protected' : '')
-                ;
-
-                $reflProperty->setAccessible(true);
-                $return->$name = self::export($reflProperty->getValue($var), $maxDepth - 1);
+        $clone = (array) $var;
+
+        foreach (array_keys($clone) as $key) {
+            $aux = explode("\0", $key);
```
Comment (Ocramius) [inline lib/Doctrine/Common/Util/Debug.php]: Nice and clean!

File: lib/Doctrine/Common/Util/Debug.php [hunk 2]
```diff
@@ -144,41 +144,26 @@ public static function export($var, $maxDepth)
...
-                $name =
-                      $reflProperty->getName()
-                    . ($return->__CLASS__ !== $currentClassName || $reflProperty->isPrivate() ? ':' . $currentClassName : '')
-                    . ($reflProperty->isPrivate() ? ':private' : '')
-                    . ($reflProperty->isProtected() ? ':protected' : '')
-                ;
-
-                $reflProperty->setAccessible(true);
-                $return->$name = self::export($reflProperty->getValue($var), $maxDepth - 1);
+        $clone = (array) $var;
```
Comment (Ocramius) [inline lib/Doctrine/Common/Util/Debug.php]: One thing that this doesn't consider is that every property is now a reference. A simple way to fix this is to use `array_map()` with a recursive identity function.

See https://3v4l.org/Yhb9u

```php
<?php

$foo = 'bar';
$bar = ['foo' => & $foo];
$baz = (object) $bar;

$baz->foo = 'tab';

var_dump($bar);
```

Comment (ostrolucky) [inline lib/Doctrine/Common/Util/Debug.php]: Cannot reproduce the problem, see new testcase

Comment (Ocramius) [inline lib/Doctrine/Common/Util/Debug.php]: Ah, sorry, missed that this is just for output, so the references are not usable anyway. Sorry :-\

**Comment by Ocramius** [comment]
Thanks @ostrolucky! Nice and clean :-)

**Pull request merged**